core/src/main/java/io/opentelemetry/android/internal/session/SessionManagerImpl.kt

```diff
@@ -57,6 +61,10 @@ internal class SessionManagerImpl(
         if (newSession != session) {
             val previousSession = session
             session = newSession
+            Context
+                .current()
+                .with(SESSION_CONTEXT_KEY, session)
+                .makeCurrent()
```

Comment:
Could this cause a context leak?

For example:

```
Span A
├─ Span B
```

If "Span B" is made current, it has a Scope that must be closed by the time it ends so that the context is restored to what it was before Span B (i.e. Span A in this case). However, if we change the current context before Span B ends, it seems like by the time its Scope is closed, it won't be able to return to Span A's context, based on [this logic](https://github.com/open-telemetry/opentelemetry-java/blob/eec21201b1b94a8838fd935b2eb5839211c5695e/context/src/main/java/io/opentelemetry/context/ThreadLocalContextStorage.java#L48).

Reply:
Yeah this is a good point that I didn't consider. I don't think that spans are relevant, but the fact that this calls `makeCurrent()` to create new scopes that never close is definitely a leak situation.